pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -165,7 +165,7 @@ func (r *ClickHouseReader) Start(readerReady chan bool) {
 	// 	Format: &allowedFormat,
 	// }
 
-	logger := promlog.New(logLevel)
+	logger := promlog.New(&promlog.Config{Level: &logLevel})
```

Comment:
This doesn't work on the develop because the promlog config change is needed for the later versions. This is why the code above is commented on. Please take a look at https://github.com/SigNoz/signoz/pull/2094, which bumps the Prometheus version and builds works on develop branch.